cpp/lib/src/MarcRecord.cc

```diff
@@ -0,0 +1,296 @@
+#include "MarcRecord.h"
+#include "util.h"
+#include "MarcReader.h"
+
+std::string MarcRecord::getFieldData(const std::string &tag) const {
+    return getFieldData(getFieldIndex(tag));
+}
+
+
+std::string MarcRecord::getFieldData(const size_t index) const {
+    if (index == MarcRecord::FIELD_NOT_FOUND || directory_entries_.cbegin() + index >= directory_entries_.cend())
+        return "";
+    const DirectoryEntry &entry(directory_entries_[index]);
+    if (not entry.hasCache()) {
```

Comment:
Do we really repeatedly use the same field data??  I think this should be optimised at the caller site.  I also think this is so ugly that it hurts. :(